simulationworkflowschema/dmft.py

```diff
@@ -17,139 +17,70 @@
 #
 from nomad.metainfo import SubSection, Quantity, Reference
 from nomad.datamodel.metainfo.simulation.method import (
-    XCFunctional, BasisSetContainer, TB as TBMethodology,
-    DMFT as DMFTMethodology
+    TB as TBMethodology,
+    DMFT as DMFTMethodology,
 )
-from nomad.datamodel.metainfo.simulation.calculation import (
-    BandGap, Dos, BandStructure, GreensFunctions
+from .general import (
+    SimulationWorkflowResults,
+    ElectronicStructureOutputs,
+    DFTMethod,
+    BeyondDFT,
 )
-from .general import SimulationWorkflowResults, SimulationWorkflowMethod, SerialSimulation
 
 
 class DMFTResults(SimulationWorkflowResults):
-    '''Groups DFT, TB and DMFT outputs: band gaps (all), DOS (DFT, TB), band
+    """
+    Groups DFT, TB and DMFT outputs: band gaps (all), DOS (DFT, TB), band
     structures (DFT, TB), Greens functions (DMFT). The ResultsNormalizer takes care
     of adding a label 'DFT', 'PROJECTION, or 'DMFT' in the method `get_dmft_workflow_properties`.
-    '''
-
-    band_gap_dft = Quantity(
-        type=Reference(BandGap),
-        shape=['*'],
-        description='''
-        DFT band gap.
-        ''')
-
-    band_gap_tb = Quantity(
-        type=Reference(BandGap),
-        shape=['*'],
-        description='''
-        TB band gap.
-        ''')
-
-    band_gap_dmft = Quantity(
-        type=Reference(BandGap),
-        shape=['*'],
-        description='''
-        DMFT band gap.
-        ''')
-
-    band_structure_dft = Quantity(
-        type=Reference(BandStructure),
-        shape=['*'],
-        description='''
-        Ref to the DFT band structure.
-        ''')
-
-    dos_dft = Quantity(
-        type=Reference(Dos),
-        shape=['*'],
-        description='''
-        Ref to the DFT density of states.
-        ''')
-
-    band_structure_tb = Quantity(
-        type=Reference(BandStructure),
-        shape=['*'],
-        description='''
-        Ref to the TB band structure.
-        ''')
-
-    dos_tb = Quantity(
-        type=Reference(Dos),
-        shape=['*'],
-        description='''
-        Ref to the TB density of states.
-        ''')
-
-    greens_functions_dmft = Quantity(
-        type=Reference(GreensFunctions),
-        shape=['*'],
-        description='''
-        Ref to the DMFT Greens functions.
-        ''')
-
-
-class DMFTMethod(SimulationWorkflowMethod):
-    '''Groups DFT, TB and DMFT input methodologies: starting XC functional, electrons
-    representation (basis set), TB method reference, DMFT method reference.
-    '''
+    """
+
+    dft_outputs = SubSection(
+        sub_section=ElectronicStructureOutputs.m_def, repeats=False
+    )
+
+    tb_outputs = SubSection(sub_section=ElectronicStructureOutputs.m_def, repeats=False)
 
-    starting_point = Quantity(
-        type=Reference(XCFunctional),
-        description='''
-        Starting point (XC functional or HF) used.
-        ''')
+    dmft_outputs = SubSection(
```

Comment:
DMFT results should inherit from ElectronicStructureOutputs also right? So the outputs are really at the top level. We do not need to put them in another sub section dmft_outputs.

Reply:
no but the context is clear here that the results belong to dmft since they are under section DMFTResults

Reply:
No, because the naming refers to the workflow, not to the single point. And the workflow only references a grouping.

Actually, this is making me wonder: in order to avoid issue, should I change the name of workflows to something to reflect better "DFT+GW" or "TB+DMFT"?

Reply:
hehe now I am confused. I have always thought that we are always referring to the dmft single point just like what we have in gw before. We do not need to define the workflow the connects the single points as this is a generic workflow but  we just set the name to something like DFT+TB+DMFT